### ISSUE TITLE
User Profiles (Use Cases)

### ISSUE THREAD

**Issue by smichel17** [mozilla/multi-account-containers#428]
I keep sharing bits and pieces of how I use containers in different issues. These pieces clog up the issues, lack context of my whole workflow, and get lost. A dedicated thread (this) solves those problems.

In this issue, **share your whole workflow, limiting yourself to one comment.** Then, link/quote your comment in other issues. If your workflow changes, edit your comment (but try to preserve the original).

**Comment by smichel17** [comment]
reserving top comment for my profile. Editing it in soon...

**Comment by jonathanKingston** [comment]
I'll share mostly because I was supposed to be writing up a **good usage of containers** style article, I can start here hah.

Despite implementing [assignment](https://jotter.jonathankingston.co.uk/blog/2017/04/04/containers-assignment/) I only just enabled the experiment on my primary browser which is Nightly today.

**Default browsing**

- My browser is configured to treat cookies as session cookies and delete all history
- I use Nightly so cookies are cleared daily
- I **never log in** to a website via a non container tab
- Search, browse MDN, use SearchFox, read web standards articles.

**Private browsing**
- As a habit from when I used to use history I still use private mode
- I use this for looking at anything:
  - I don't trust
  - I would never want people knowing that I search for...
    - When you hear something on twitter and you be like "err who is Kendal Jenner anyways?!" and know for a fact it's some pointless person.
    - When you hear a lyric in your head but are not sure if you should be ashamed of yourself

I wish these were first party isolated for all my use cases this would be ok.

**Work**
- At work we use oauth for most services so having a work container makes 100% sense here
  - Mail, Calendar, Wiki pages, Other stuff that really shouldn't be private but would bore the public to death if it was public
  - Mostly work things all are clumped in here even if they are public (etherpad)
  - We use Google services at Mozilla (we are not secret about that) and well Google is baked into our oauth provider so 50/60% of my work websites can't be protected much from GA tracking besides maybe logging into via oauth multiple times etc
  - Since I have enabled assignment 40-50% of my tabs which are Bugzilla could be converted into being a container on it's own as that doesn't use oauth. This would allow me to hide Bugzilla sometimes.

**Personal**
- Mail
- Twitter

**Pinned tabs** (each with a unique container)
- Music (assigned tab)
- Password manager (assigned tab)
- Mail provider 2 (assigned tab)
- Messaging app (assigned tab)

For pinned tabs, these also could be first party isolated and not in my containers menu.

**Local domains**
- I have a container for my routers at work, home that I don't want public internet knowing about if I can help it.

**GitHub**
- As of today I created a GitHub account which was in personal as an assigned domain

**Comment by grahamperrin** [comment]
I'm here via https://twitter.com/KingstonTime/status/850261536828399616 thanks @jonathanKingston 

The best introduction to my use cases might be: 

- [The window is the container – the user groups things within a container – those things are tabs](https://discourse.mozilla-community.org/t/the-window-is-the-container-the-user-groups-things-within-a-container-those-things-are-tabs/15451?u=grahamperrin) (2017-05-06)

– with [post 3](https://discourse.mozilla-community.org/t/could-panorama-feature-be-brought-back-to-firefox/14672/3?u=grahamperrin) under _Could PANORAMA feature be brought back to Firefox?_ as background, in particular: 

> … twelve desktops, …

## tl;dr 

Visually, from three different dates: 

![twelve desktops](https://cloud.githubusercontent.com/assets/192271/25773919/55aea396-327e-11e7-9efa-d5fdaedb8640.png)

![tab groups one group of tabs](https://cloud.githubusercontent.com/assets/192271/25773922/5d5a7746-327e-11e7-8cbd-651b121dd926.png)

![windows as containers containers within a panel of kde plasma 4](https://cloud.githubusercontent.com/assets/192271/25773924/6721829c-327e-11e7-9aa7-c86d5656bab1.png)

If you have any questions, please post under https://discourse.mozilla-community.org/t/-/15451 – I'll add some more screenshots there, with reference to this issue `#428`.

Thanks

**Comment by grahamperrin** [comment]
A more recent use case: 

![2017-08-28 21 57 17](https://user-images.githubusercontent.com/192271/29794342-7e4db466-8c3f-11e7-9eb6-42a114fb2b01.png)

– in a virtual desktop, with a profile that's separate from my main Firefox profile, one container for each of the five e-mail addresses that I'm required to use at work. 

The five containers allow me to run five copies of Outlook Web App (one for each user account) in a single window. 

Bottom right, the new window could not be contained (#770). 

Behind that window (top left), the colour is wrong (#772). 

I don't like the clutter. I'd prefer my containers to be colourless (#391) and free from icons (#771).

**Comment by grahamperrin** [comment]
Another desktop Firefox use case: 

## Terse ratings at `addons.mozilla.org` (AMO)

### Preparations

1. install [Context Plus](https://addons.mozilla.org/addon/context-plus/)
2. install [Switch Container](https://addons.mozilla.org/addon/switch-container/)
3. in a (default) non-contained tab
4. at any AMO page, log in
5. create a container, _mobile addons.mozilla.org_
6. open a tab in _mobile addons.mozilla.org_
7. **again**, at any AMO page, log in
8. click 'View Mobile Site'
9. use Context Plus to reopen the page in _default_. 

### Routine

1. visit the AMO page of the add-on that you'd like to rate
2. use Switch Container to reopen the page in _mobile addons.mozilla.org_
3. give your star rating
4. escape from the _written review_ dialogue
5. use Context Plus to reopen the page in _default_. 

### Why use Switch Container?

Because a new container may be not immediately available to Context Plus. 

### Why use Context Plus? 

Because [Switch Container can not switch to default](https://gitlab.com/mjanetmars/switch-container/issues/3).

**Comment by grahamperrin** [comment]
Also I often use: 

- a single container across multiple windows

– and ([Outlook Web App x5](https://github.com/mozilla/multi-account-containers/issues/428#issuecomment-325489803) above as an example) at least one window **requires** visibility of multiple containers, and non-contained pages, within that window.

**Comment by grahamperrin** [comment]
From the day when I first noticed a thousand tabs in a session: 

![2017-10-09 05 57 31](https://user-images.githubusercontent.com/192271/31647927-fb1bcab6-b302-11e7-9a1b-bd2765dbede3.png)

In the screen to the left: 288 tabs in my _Mozilla_ window (the name is held in mind, not visible). 

Above the list of grouped tabs, the **visible name** of the group: 

- containers

(Coincidence: I hit the 1,000-tab mark when I opened a tab to a page about containers.)

Around an hour later I captured some of the session data to a PDF. A version that includes groups (intentionally omitting titles and URLs of open tabs) is at <https://github.com/maheee/mHeaderControl/issues/1#issuecomment-337119354>. Probably seventy-something groups. 

To the left of the list of tabs: in a panel of KDE, my list of windows. Firefox had around fourteen. The visible _mHeaderControl …_ is a page within what I think of as my _extensions_ window. I expect this window to be my busiest, from a group numbers perspective, for the next year or so. 

----

Now, sixty-something groups in that window, each group almost completely devoted to a single extension (the group near the center is much looser, a _mash_): 

![2017-10-17 06 37 45](https://user-images.githubusercontent.com/192271/31648796-a08a4be0-b307-11e7-9f36-97719b49ec9f.png)

– and this page is one of thirty tabs within my _Multi-Account Containers_ group: 

![2017-10-17 06 39 01](https://user-images.githubusercontent.com/192271/31648813-ad18adac-b307-11e7-8d5c-3bd9c1ce0153.png)

With this seventy-something-group profile, I use one container: 

- to occasionally preview the new `addons.mozilla.org`.

**Comment by grahamperrin** [comment]
I **can't understand _why_** this seems to work, neither is it privacy-related, but this **use of Firefox containers** – as a workaround to a bug that can affect safe mode of Firefox without containers – is worth mentioning: 

- **[Gitter](https://gitter.im/)-oriented [post 2](https://discourse.mozilla.org/t/aiming-to-work-with-no-more-than-one-container--and--avoid-default-non-contained-work/20025/2?u=grahamperrin)** under _Aiming to work with no more than one container _**and**_ avoid (default) non-contained work - … Containers - Mozilla Discourse_

**Comment by practik** [comment]
I'm primarily interested in limiting Google's tracking of my searches. I block cookies from `google.com`, but to be able to use Google Books effectively I must allow cookies from `books.google.com`. Unfortunately, Books pages are somehow able to set `google.com` cookies on me even though such cookies are otherwise blocked.

I had hoped to solve this by doing my basic searches in a dedicated Container and automatically (with the help of a user script) opening Books links in a new tab and different Container. Unfortunately, at the moment those links open new tabs in the **same** Container. I'd very much like to see a pref added to control this as discussed at #434.

**Comment by jonathanKingston** [comment]
I'm going to close this off, I don't think we are doing anything with it and it hasn't moved on since November too. People can continue adding comments though; thanks for raising this.
